drivers/mspi/mspi_dw_vendor_specific.h

```diff
@@ -97,7 +101,217 @@ static inline int vendor_specific_xip_disable(const struct device *dev,
+			 BIT(EVDMA_STATIC_ADDR) | BIT(EVDMA_PLAIN_DATA_BUF_WR))
+
+
+typedef struct {
+	uint8_t *addr;
+	uint32_t attr;
+} EVDMA_JOB_Type;
+
+#define EVDMA_JOB(BUFFER, SIZE, ATTR) \
+	(EVDMA_JOB_Type) { .addr = (uint8_t *)BUFFER, .attr = (ATTR << EVDMA_ATTR_ATTR_Pos | SIZE) }
+#define EVDMA_NULL_JOB() \
+	(EVDMA_JOB_Type) { .addr = (uint8_t *)0, .attr = 0 }
+typedef struct {
+	EVDMA_JOB_Type *tx_job;
+	EVDMA_JOB_Type *rx_job;
+} QSPI_TRANSFER_LIST_Type;
+
+/* Number of jobs needed for transmit trasaction */
+#define MAX_NUM_JOBS 4
+
+/* Vendor-specific data structure for Nordic QSPI */
+typedef struct {
+	QSPI_TRANSFER_LIST_Type *transfer_list;
+	EVDMA_JOB_Type *joblist;
+} nordic_qspi_vendor_data_t;
+
+/* Static allocation macros for vendor-specific data */
+#define VENDOR_SPECIFIC_DATA_DEFINE(inst) \
+	static QSPI_TRANSFER_LIST_Type mspi_dw_##inst##_transfer_list; \
+	static EVDMA_JOB_Type mspi_dw_##inst##_joblist[MAX_NUM_JOBS]; \
+	static nordic_qspi_vendor_data_t mspi_dw_##inst##_vendor_data = { \
+		.transfer_list = &mspi_dw_##inst##_transfer_list, \
+		.joblist = &mspi_dw_##inst##_joblist[0] \
+	};
+
+#define VENDOR_SPECIFIC_DATA_GET(inst) &mspi_dw_##inst##_vendor_data
+
+/* Temporarily hard-coded as not in MDK yet */
+#define QSPI_TMOD_OFFSET (0x490UL)
+#define QSPI_TMOD_RX_ONLY (0x2)
+static inline void vendor_specific_start_dma_xfer(const struct device *dev)
+{
+	struct mspi_dw_data *dev_data = dev->data;
+	const struct mspi_dw_config *config = dev->config;
+	const struct mspi_xfer_packet *packet =
+		&dev_data->xfer.packets[dev_data->packets_done];
+	NRF_QSPI_Type *preg = (NRF_QSPI_Type *)config->wrapper_regs;
+
+	/* Use vendor-specific data from config - stores job and transfer lists */
+	nordic_qspi_vendor_data_t *vendor_data = (nordic_qspi_vendor_data_t *)
+						 config->vendor_specific_data;
+
+
+	QSPI_TRANSFER_LIST_Type *transfer_list = vendor_data->transfer_list;
+	EVDMA_JOB_Type *joblist = vendor_data->joblist;
+
+	int tmod = 0;
+	int job_idx = 0;
+
+	if (packet->dir == MSPI_TX) {
+		preg->CONFIG.RXTRANSFERLENGTH = 0;
+
+		/* Setting up EVDMA joblist depending on cmd, addr and data */
+
+		/*
+		 * Command address will always have a length of 4 from the DMA's perspective,
+		 * QSPI peripheral will use length of data specified in core registers
+		 */
+		if (dev_data->xfer.cmd_length > 0) {
+			joblist[job_idx++] = EVDMA_JOB(&packet->cmd, 4, EVDMA_ATTRIBUTE);
+		}
+		if (dev_data->xfer.addr_length > 0) {
+			joblist[job_idx++] = EVDMA_JOB(&packet->address, 4, EVDMA_ATTRIBUTE);
+		}
+		if (packet->num_bytes > 0) {
+			joblist[job_idx++] = EVDMA_JOB(packet->data_buf, packet->num_bytes,
+						       EVDMA_ATTRIBUTE);
+		}
+		/* Always terminate with null job */
+		joblist[job_idx] = EVDMA_NULL_JOB();
+		/* tx_job should point to first valid job, or null if none */
+		if (job_idx > 0) {
+			transfer_list->tx_job = &joblist[0];
+		} else {
+			transfer_list->tx_job = &joblist[job_idx];
+		}
```

Comment:
tx_job cannot be empty, see: https://github.com/Hi-Im-David/zephyr/blob/6875795cd4d34e8831873972cbc2c286dcd24b6b/drivers/mspi/mspi_dw.c#L1097-L1101